File: mobile/src/api/recurring.ts
import apiClient from './client';
import type { RecurringTransaction } from '@/types';

interface RecurringCreate {
  account_id: number;
  category_id?: number;
  amount: number;
  description: string;
  frequency: string;
  transaction_type: string;
  next_date: string;
  end_date?: string;
}

interface RecurringUpdate {
  amount?: number;
  description?: string;
  frequency?: string;
  next_date?: string;
  end_date?: string;
  is_active?: boolean;
}

export const recurringApi = {
  list: async (): Promise<RecurringTransaction[]> => {
    const response = await apiClient.get<RecurringTransaction[]>('/recurring/');
    return response.data;
  },

  create: async (data: RecurringCreate): Promise<RecurringTransaction> => {
    const response = await apiClient.post<RecurringTransaction>(
      '/recurring/',
      data,
    );
    return response.data;
  },

  update: async (
    id: number,
    data: RecurringUpdate,
  ): Promise<RecurringTransaction> => {
    const response = await apiClient.patch<RecurringTransaction>(
      `/recurring/${id}`,
      data,
    );
    return response.data;
  },

  delete: async (id: number): Promise<void> => {
    await apiClient.delete(`/recurring/${id}`);
  },
};

File: backend/app/schemas/recurring.py
from pydantic import BaseModel
from datetime import date, datetime


class RecurringTransactionCreate(BaseModel):
    account_id: int
    category_id: int | None = None
    amount: float
    description: str
    frequency: str
    transaction_type: str
    next_date: date
    end_date: date | None = None


class RecurringTransactionUpdate(BaseModel):
    amount: float | None = None
    description: str | None = None
    frequency: str | None = None
    next_date: date | None = None
    end_date: date | None = None
    is_active: bool | None = None


class RecurringTransactionResponse(BaseModel):
    id: int
    account_id: int
    category_id: int | None
    amount: float
    description: str
    frequency: str
    transaction_type: str
    next_date: date
    end_date: date | None
    is_active: bool
    created_at: datetime

    model_config = {"from_attributes": True}
